restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/client/rcmlserver/RcmlserverApi.java

```diff
@@ -35,27 +33,43 @@
 import org.restcomm.connect.commons.util.SecurityUtils;
 import org.restcomm.connect.commons.util.UriUtils;
 import org.restcomm.connect.dao.entities.Account;
-import org.restcomm.connect.http.exceptions.RcmlserverNotifyError;
 
-import java.io.IOException;
 import java.net.URI;
 import java.net.URISyntaxException;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import org.apache.http.client.config.RequestConfig;
 import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.concurrent.FutureCallback;
+import org.apache.http.impl.nio.client.CloseableHttpAsyncClient;
 import org.apache.http.protocol.BasicHttpContext;
 import org.apache.http.protocol.HttpContext;
+import org.restcomm.connect.dao.entities.Account.Status;
 
 /**
- * Utility class that handles notification submission to rcmlserver (typically RVD)
+ * Utility class that handles notification submission to rcmlserver (typically
+ * RVD)
  *
  * @author otsakir@gmail.com - Orestis Tsakiridis
  */
 public class RcmlserverApi {
+
     static final Logger logger = Logger.getLogger(RcmlserverApi.class.getName());
 
     enum NotificationType {
-        accountClosed
+        accountClosed, accountSuspended,
```

Comment:
@jaimecasero Apparently RVD needs to get notified for all state changes in the accounts, right?

Reply:
Well, i was checking the code before @otsakir  answered my challenge, and decided this may be applicable not only to CLOSED, but all states. Then orestis finally answered, and we decided to keep the notification only for CLOSED status. Anyway,i left the new implementation,as is using async http client, and we may potentially send notification about any status (more generic approach)